Mathlib/Combinatorics/SimpleGraph/Acyclic.lean

```diff
@@ -197,4 +198,33 @@ lemma IsTree.card_edgeFinset [Fintype V] [Fintype G.edgeSet] (hG : G.IsTree) :
       refine (hG.existsUnique_path _ _).unique ((hf _).takeUntil _) ?_
       simp [h.ne]
 
+/-- A minimally connected graph is a tree. TODO : Iff version. -/
+lemma isTree_of_minimal_connected (h : Minimal Connected G) : IsTree G := by
+  rw [isTree_iff, and_iff_right h.prop, isAcyclic_iff_forall_adj_isBridge]
+  exact fun _ _ _ ↦ by_contra fun hbr ↦ h.not_prop_of_lt (by simpa [← edgeSet_ssubset_edgeSet])
+    <| h.prop.connected_delete_edge_of_not_isBridge hbr
+
+/-- Every connected graph has a spanning tree. TODO : remove the `Fintype V` hypothesis. -/
+lemma Connected.exists_isTree_le [Fintype V] (h : G.Connected) : ∃ T ≤ G, IsTree T := by
+  obtain ⟨T, hTG, hmin⟩ := {H : SimpleGraph V | H.Connected}.toFinite.exists_minimal_le h
+  exact ⟨T, hTG, isTree_of_minimal_connected hmin⟩
+
+/-- Every connected graph on `n` vertices has at least `n-1` edges. -/
+lemma Connected.card_vert_le_card_edgeSet_add_one [Fintype V] [Fintype G.edgeSet]
+    (h : G.Connected) : Fintype.card V ≤ Fintype.card G.edgeSet + 1 := by
```

Comment:
As above, we should prefer `Finite` over `Fintype`.

In this case we need to change the statement to avoid needing `Fintype`: we switch to `Nat.card`. This will require adding the import `import Mathlib.SetTheory.Cardinal.Finite` but as `SimpleGraph.Acyclic` is only imported by one file, which is a leaf file, this is an acceptable import change.
```suggestion
lemma Connected.card_vert_le_card_edgeSet_add_one [Finite V]
    (h : G.Connected) : Nat.card V ≤ Nat.card G.edgeSet + 1 := by
```
Admittedly some of the other lemmas in this file currently use the inferior `Fintype` spelling but this inconsistency is the case throughout the library: we are gradually shifting as much as make sense to `Finite`.